### PR TITLE
 Rename Test_onVmStart to Test_pluginInitialization

### PR DESCRIPTION
**Pull request by anuraaga**
This came to mind in #327 but wasn't sure about it so figured a separate PR is better

### CODE REVIEW

File: proxywasm/internal/abi_callback_configuration_test.go [hunk 1]
```diff
@@ -65,8 +66,15 @@ func Test_proxyOnVMStart(t *testing.T) {
...
-	require.NotNil(t, proxyOnMemoryAllocate(100))
+	// Allocate memory. The returned buffer has the same size as passed in
+	bufPtr := proxyOnMemoryAllocate(100)
+	buf := unsafe.Slice(bufPtr, 101)
+	for _, b := range buf[:100] {
+		require.Equal(t, byte(0), b)
+	}
+	require.Panics(t, func() {
+		_ = buf[101]
+	})
```
Comment (mathetake) [inline proxywasm/internal/abi_callback_configuration_test.go:77]: can we move this out of `Test_proxyOnVMStart` into a separate `Test_proxyOnMemoryAllocate`?

File: proxywasm/internal/abi_callback_configuration_test.go [hunk 2]
```diff
@@ -65,8 +66,15 @@ func Test_proxyOnVMStart(t *testing.T) {
 	// Check ABI version. There is no return value so just make sure it doesn't panic.
 	proxyABIVersion()
```
Comment (mathetake) [inline proxywasm/internal/abi_callback_configuration_test.go:67]: this as well

File: proxywasm/internal/abi_callback_configuration_test.go (hunk 1 above)
Comment (anuraaga) [reply proxywasm/internal/abi_callback_configuration_test.go:77]: What do you think about renaming `Test_proxyOnVMStart` to `Test_pluginInitialization`? I noticed there are several ABI tested for initialization in this function so adding the other ones that Envoy would call felt natural. Or I can separate out no worries, just want to doublecheck

Comment (mathetake) [reply proxywasm/internal/abi_callback_configuration_test.go:77]: works for me!

Comment (anuraaga) [reply proxywasm/internal/abi_callback_configuration_test.go:77]: Thanks, renamed

File: proxywasm/internal/abi_callback_configuration_test.go [hunk 3]
```diff
@@ -57,16 +58,23 @@ func Test_proxyOnVMStart(t *testing.T) {
...
-	// Allocate memory
-	require.NotNil(t, proxyOnMemoryAllocate(100))
+	// Allocate memory. The returned buffer has the same size as passed in
+	bufPtr := proxyOnMemoryAllocate(100)
+	buf := unsafe.Slice(bufPtr, 101)
+	for _, b := range buf[:100] {
+		require.Equal(t, byte(0), b)
+	}
+	require.Panics(t, func() {
+		_ = buf[101]
```
Comment (mathetake) [inline proxywasm/internal/abi_callback_configuration_test.go:76]: shouldn't this be 100?
```suggestion
		_ = buf[100]
```

**Review by mathetake — APPROVED** [review]

File: proxywasm/internal/abi_callback_configuration_test.go [hunk 4]
```diff
@@ -57,16 +58,23 @@ func Test_proxyOnVMStart(t *testing.T) {
...
 	proxyOnVMStart(0, 0)
 	require.True(t, vmContext.onVMStartCalled)
 	require.Equal(t, uint32(0), currentState.activeContextID)
 
-	// Check ABI version. There is no return value so just make sure it doesn't panic.
-	proxyABIVersion()
-
-	// Allocate memory
-	require.NotNil(t, proxyOnMemoryAllocate(100))
+	// Allocate memory. The returned buffer has the same size as passed in
```
Comment (mathetake) [inline proxywasm/internal/abi_callback_configuration_test.go:69]: I think this is something untestable, isn't it

Comment (anuraaga) [reply proxywasm/internal/abi_callback_configuration_test.go:69]: I thought Go would verify the memory access but maybe I've been in wasm too long 😂 Yeah `unsafe` removes all bets here I guess. I will just rename the test I guess

**Pull request merged**